RingPublishingTracking/src/main/java/com/ringpublishing/tracking/internal/decorator/UserIdentifierDataDecorator.kt

```diff
@@ -61,9 +62,14 @@ internal class UserIdentifierDataDecorator(
         }
     }
 
-    private fun encodeUserData(sso: Sso?, artemisId: ArtemisId?): String?
-    {
-        val jsonUser = gson.toJson(UserIdentifier(artemisId, sso))
+    private fun encodeUserData(sso: Sso?, artemisId: ArtemisId?, isActiveSubscriber: Boolean?): String? {
+        val jsonUser = gson.toJson(
+            UserIdentifier(
+                id = artemisId,
+                sso = sso,
+                type = if (isActiveSubscriber == true) "subscriber" else null
```

Comment:
Może if (isActiveSubscriber == true) "subscriber" else null to przenieść już do środka UserIdentifier.